sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1986,66 +2015,27 @@ pi_result piDevicePartition(pi_device Device,
 
   PI_ASSERT(Device, PI_INVALID_DEVICE);
 
-  // Check if Device was already partitioned into the same or bigger size
-  // before. If so, we can return immediately without searching the global
-  // device cache. Note that L0 driver always returns the same handles in the
-  // same order for the given number of sub-devices.
-  if (OutDevices && NumDevices <= Device->SubDevices.size()) {
-    for (uint32_t I = 0; I < NumDevices; I++) {
-      OutDevices[I] = Device->SubDevices[I];
-      // reusing the same pi_device needs to increment the reference count
-      piDeviceRetain(OutDevices[I]);
-    }
-    if (OutNumDevices)
-      *OutNumDevices = NumDevices;
-    return PI_SUCCESS;
+  // Devices cache is normally created in piDevicesGet but still make
+  // sure that cache is populated.
+  //
+  pi_result Res = populateDeviceCacheIfNeeded(Device->Platform);
+  if (Res != PI_SUCCESS) {
+    return Res;
   }
 
-  // Get the number of subdevices available.
-  // TODO: maybe add interface to create the specified # of subdevices.
-  uint32_t Count = 0;
-  ZE_CALL(zeDeviceGetSubDevices, (Device->ZeDevice, &Count, nullptr));
-
   if (OutNumDevices) {
-    *OutNumDevices = Count;
+    *OutNumDevices = Device->SubDevices.size();
```

Comment:
It looks to me here that we always return all maximum number of sub-devices regardless of user's request.
Is this right?